File: src/models/SelectedCourse.java
package models;
import enums.CourseStatus;

public class SelectedCourse {
    private Student student;
    private Course course;
    private CourseStatus status;
    private CourseSection courseSection;

    public SelectedCourse(Student student, Course course, CourseSection courseSection) {
        this.student = student;
        this.course = course;
        this.status = CourseStatus.DRAFT;
        this.courseSection = courseSection;
    }

    public void setStatus(CourseStatus status) {
        this.status = status;
    }

    public boolean isApproved() {
        if (status == CourseStatus.APPROVED) {
            return true;
        } 
        else {
            return false;
        }
    }

    public Student getStudent() {
        return student;
    }

    public Course getCourse() {
        return course;
    }

    public CourseStatus getStatus() {
        return status;
    }

    public CourseSection getCourseSection() {
        return courseSection;
    }
}

File: iteration3/src/models/SelectedCourse.py
import logging
from iteration3.src.enums.CourseStatus import CourseStatus
from iteration3.src.models.Course import Course
from iteration3.src.models.CourseSection import CourseSection

class SelectedCourse(object):
    def __init__(self,course,status,courseSection):
        self._course = course
        self._status = status
        self._courseSection = courseSection

    @classmethod
    def dictToObject(cls, dict):
        return cls(
            Course.dictToObject(dict['course']),
            CourseStatus.dictToEnum(dict['status']),
            CourseSection.dictToObject(dict['courseSection'])
        )


    def setStatus(self,status):
        self._status = status
        logging.log(logging.INFO, f"Course ({self._course.getCourseCode()}) status changed to: {status}")

    def getCourse(self):
        return self._course
    
    def getCourseSection(self):
        return self._courseSection

    def getStatus(self):
        return self._status
    
    def setStatus(self, status):
        self._status = status
            

